src/AuctionHouseBotConfig.cpp

```diff
@@ -0,0 +1,318 @@
+/*
+ * This file is part of the AzerothCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by the
+ * Free Software Foundation; either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AuctionHouseBotConfig.h"
+#include "AuctionHouseMgr.h"
+#include "ItemTemplate.h"
+#include <numeric>
+
+AHBConfig::AHBConfig(uint32 ahid)
+{
+    _auctionHouseID = ahid;
+
+    switch (ahid)
+    {
+    case AUCTIONHOUSE_ALLIANCE:
+        _auctionHouseFactionID = 55;
+        break;
+    case AUCTIONHOUSE_HORDE:
+        _auctionHouseFactionID = 29;
+        break;
+    case AUCTIONHOUSE_NEUTRAL:
+        _auctionHouseFactionID = 120;
+        break;
+    default:
+        _auctionHouseFactionID = 120;
+        break;
+    }
+}
+
+uint32 AHBConfig::GetMinItems()
+{
+    if (!_minItems && _maxItems)
+        return _maxItems;
+    else if (_maxItems && (_minItems > _maxItems))
+        return _maxItems;
+
+    return _minItems;
+}
+
+void AHBConfig::SetPercentages(std::array<uint32, AHB_MAX_QUALITY>& percentages)
+{
+    uint32 totalPercent = std::accumulate(percentages.begin(), percentages.end(), 0);
+
+    if (totalPercent == 0)
+    {
+        _maxItems = 0;
+    }
+    else if (totalPercent != 100)
+    {
+        _itemsPercent[ITEM_QUALITY_POOR]            = 0;
```

Comment:
Should print a warning log message here?

Or re-normalize the values

80+80 = 160
value*(100/160)
so it becomes 50+50